src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -129,33 +154,52 @@ private OnnxTransform(IHostEnvironment env, Arguments args, byte[] modelBytes =
                 Model = OnnxModel.CreateFromBytes(modelBytes);
 
             var modelInfo = Model.ModelInfo;
-            if (modelInfo.InputsInfo.Length != 1)
-                throw env.Except($"OnnxTransform supports Onnx models with one input. The provided model has ${modelInfo.InputsInfo.Length} input(s).");
-            if (modelInfo.OutputsInfo.Length != 1)
-                throw env.Except($"OnnxTransform supports Onnx models with one output. The provided model has ${modelInfo.OutputsInfo.Length} output(s).");
-
-            Input = args.InputColumn;
-            Output = args.OutputColumn;
-
-            var outputNodeInfo = Model.ModelInfo.OutputsInfo[0];
-            var type = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
-            var shape = outputNodeInfo.Shape;
-            var dims = shape.Count > 0 ? shape.Skip(shape[0] < 0 ? 1 : 0).Select( x => (int) x ).ToArray() : new[] { 0 };
-            OutputType = new VectorType(type, dims);
+            //if (modelInfo.InputsInfo.Length != 1)
+            //    throw env.Except($"OnnxTransform supports Onnx models with one input. The provided model has ${modelInfo.InputsInfo.Length} input(s).");
+            //if (modelInfo.OutputsInfo.Length != 1)
+            //    throw env.Except($"OnnxTransform supports Onnx models with one output. The provided model has ${modelInfo.OutputsInfo.Length} output(s).");
+
+            Inputs = args.InputColumns;
+            Outputs = args.OutputColumns;
+            //var type = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
+            //var shape = outputNodeInfo.Shape;
+            //var dims = shape.Count > 0 ? shape.Skip(shape[0] < 0 ? 1 : 0).Select( x => (int) x ).ToArray() : new[] { 0 };
```

Comment:
delete as well?

Reply:
Yes, this can be deleted.

---
In reply to: [233267902](https://github.com/dotnet/machinelearning/pull/1586#discussion_r233267902) [](ancestors = 233267902)